@vates/types/src/common.mts

```diff
@@ -0,0 +1,638 @@
+declare const __brand: unique symbol
+
+export type Branded<TBrand extends string, TType = string> = TType & { [__brand]: TBrand }
+
+export const TASK_ALLOWED_OPERATIONS = {
+  CANCEL: 'cancel',
+  DESTROY: 'destroy',
+} as const
+
+export type TASK_ALLOWED_OPERATIONS = (typeof TASK_ALLOWED_OPERATIONS)[keyof typeof TASK_ALLOWED_OPERATIONS]
+
+export const TASK_STATUS_TYPE = {
+  CANCELLED: 'cancelled',
+  CANCELLING: 'cancelling',
+  FAILURE: 'failure',
+  PENDING: 'pending',
+  SUCCESS: 'success',
+} as const
```

Comment:
why does we use const twice ? 
export const name =  {} as const ?

Reply:
With `as const` the type will be:

```ts
const TASK_STATUS_TYPE: {
    readonly CANCELLED: "cancelled";
    readonly CANCELLING: "cancelling";
    readonly FAILURE: "failure";
    readonly PENDING: "pending";
    readonly SUCCESS: "success";
}
```

Without, it would have been:

```ts
const TASK_STATUS_TYPE: {
    CANCELLED: string;
    CANCELLING: string;
    FAILURE: string;
    PENDING: string;
    SUCCESS: string;
}
```